src/tools/wasm-as.cpp

```diff
@@ -51,6 +53,12 @@ int main(int argc, const char *argv[]) {
       .add("--debuginfo", "-g", "Emit names section and debug info",
            Options::Arguments::Zero,
            [&](Options *o, const std::string &arguments) { debugInfo = true; })
+      .add("--source-map", "-sm", "Emit source map to the specified file",
+           Options::Arguments::One,
+           [&sourceMapFilename](Options *o, const std::string &argument) { sourceMapFilename = argument; })
+      .add("--source-map-url", "-su", "Use specified string as source map URL",
+           Options::Arguments::One,
+           [&sourceMapUrl](Options *o, const std::string &argument) { sourceMapUrl = argument; })
```

Comment:
I would rather move this out of scope this PR -- `--source-map` and `--source-map-url` are not really optimization options and have different meaning in some cases, e.g. for asm2wasm `--source-map` means output file, but for wasm-dis -- input file.